exchange/exchange.go

```diff
@@ -497,6 +508,30 @@ func (e *exchange) getAllBids(
 	return adapterBids, adapterExtra, bidsFound
 }
 
+func insertStoredImpData(impsToStoredRequest map[string][]byte, bids *pbsOrtbSeatBid) error {
+	for _, bid := range bids.bids {
+		if val, ok := impsToStoredRequest[bid.bid.ImpID]; ok {
+			bidExtData := make(map[string]interface{})
+			storedImpVal := make(map[string]interface{})
+			err := json.Unmarshal(bid.bid.Ext, &bidExtData)
+			if err != nil {
+				return err
+			}
+			err = json.Unmarshal(val, &storedImpVal)
+			if err != nil {
+				return err
+			}
+			bidExtData[StoredRequestAttributes] = storedImpVal
+			result, err := json.Marshal(bidExtData)
+			if err != nil {
+				return err
+			}
+			bid.bid.Ext = result
```

Comment:
Looks good. With the code as it is now, you don't need to unmarshal the stored request. This could be simplified as:
```
bidExtData := make(map[string]json.RawMessage)
err := json.Unmarshal(bid.bid.Ext, &bidExtData)
if err != nil {
	return err
}
bidExtData[StoredRequestAttributes] = val
result, err := json.Marshal(bidExtData)
if err != nil {
	return err
}
bid.bid.Ext = result
```

This approach may also be faster since `interface{}` forces type discovery and `json.RawMessage` does not.

Anyway, this may be a moot point as you'll likely need to unmarshal the stored request - at least partially - to accomodate @mansinahar's comment.

Reply:
Yes, it works! Modified